lighthouse-core/config/config.js

```diff
@@ -330,6 +319,121 @@ class Config {
     return merge(baseJSON, extendJSON);
   }
 
+ /**
+  * Filter out any unrequested aggregations from the config. If any audits are
+  * no longer needed by any remaining aggregations, filter out those as well.
+  * @param {!Object} oldConfig Lighthouse config object.
+  * @param {!Array<string>} aggregationNames Name values of aggregations to include.
+  * @return {undefined} config is mutated, instead of returning
+  */
+  static generateNewConfigOfAggregations(oldConfig, aggregationNames) {
+    // 0. Clone config to avoid mutating it
+    const config = JSON.parse(JSON.stringify(oldConfig));
+    // 1. Filter to just the chosen aggregations
+    config.aggregations = config.aggregations.filter(agg => aggregationNames.includes(agg.name));
+
+    // 2. Resolve which audits will need to run
+    const requestedAuditNames = Config.getAuditsNeededByAggregations(config.aggregations);
+    const auditPathToNameMap = Config.getMapOfAuditPathToName(config);
+    config.audits = config.audits.filter(auditPath =>
+        requestedAuditNames.has(auditPathToNameMap.get(auditPath)));
+
+    // 3. Resolve which gatherers will need to run
+    const auditObjectsSelected = Config.requireAudits(config.audits);
+    const requiredGatherers = Config.getGatherersNeededByAudits(auditObjectsSelected);
+
+    // 4. Filter to only the neccessary passes
+    config.passes = Config.getPassesNeededByGatherers(config.passes, requiredGatherers);
+    return config;
+  }
+
+ /**
+  * Return names of top-level aggregations from a config. Used by tools driving lighthouse-core
+  * @param {!Object} config Lighthouse config object.
+  * @return {!Array<string>}  Name values of aggregations within
+  */
+  static getAggregationNames(config) {
+    return config.aggregations.map(agg => agg.name);
+  }
+
+  /**
+   * Find audits required for remaining aggregations.
+   * @param  {!Object} aggregations
+   * @return {!Set<string>}
+   */
+  static getAuditsNeededByAggregations(aggregations) {
+    const requestedItems = _flatten(aggregations.map(aggregation => aggregation.items));
+    const requestedAudits = _flatten(requestedItems.map(item => Object.keys(item.audits)));
+    return new Set(requestedAudits);
+  }
+
+  /**
+   * Creates mapping from audit path (used in config.audits) to audit.name (used in config.aggregations)
+   * @param {!Object} config Lighthouse config object.
+   * @return {Map}
+   */
+  static getMapOfAuditPathToName(config) {
+    // The `audits` property in the config is a list of paths of audits to run.
+    // `requestedAuditNames` is a list of audit *names*. Map paths to names, then
```

Comment:
`requestedAuditNames` isn't a thing anymore, so need to update comment. Maybe just `config.audits` instead?